### PR TITLE
feat(compute): compute_spot_create sample

### PR DESCRIPTION
**Pull request by irudykss**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/instances/spot/spot_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	defer func() {
+		if err := deleteInstance(ctx, tc.ProjectID, zone, instanceName); err != nil {
+			t.Errorf("deleteInstance got err: %v", err)
+		}
+
+	}()
+
+	buf.Reset() // to check list output
+
+	err = listInstances(buf, tc.ProjectID, zone)
```
Comment (muncus) [inline compute/instances/spot/spot_test.go:106]: Why list the instances, when you could just Get()? https://cloud.google.com/go/docs/reference/cloud.google.com/go/compute/latest/apiv1#cloud_google_com_go_compute_apiv1_InstancesClient_Get

File: compute/instances/spot/create_spot.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	instance, err := createInstance(ctx, projectID, zone, instanceName, disks, "n1-standard-1", "global/networks/default", true)
+	if err != nil {
+		return fmt.Errorf("createInstance: %w", err)
+	}
+
+	fmt.Fprintf(w, "Instance created: %v\n", instance)
+	return nil
+}
+
+func main() {
```
Comment (muncus) [inline compute/instances/spot/create_spot.go:142]: snippets should not have a main function. please remove this.

**Review by muncus — CHANGES_REQUESTED** [review]

File: compute/instances/spot/spot_test.go (hunk 1 above)
Comment (irudykss) [reply compute/instances/spot/spot_test.go:106]: fixed and moved to [PR](https://github.com/GoogleCloudPlatform/golang-samples/pull/4187)

File: compute/instances/spot/create_spot.go (hunk 2 above)
Comment (irudykss) [reply compute/instances/spot/create_spot.go:142]: fixed and moved to [PR](https://github.com/GoogleCloudPlatform/golang-samples/pull/4187)

**Comment by irudykss** [comment]
Sample moved to combined [PR](https://github.com/GoogleCloudPlatform/golang-samples/pull/4187)

**Comment by irudykss** [comment]
Sample moved to combined [PR](https://github.com/GoogleCloudPlatform/golang-samples/pull/4187)

**Pull request closed without merging**